config/initializers/cookie_rotator.rb

```diff
@@ -0,0 +1,14 @@
+Rails.application.config.after_initialize do
+  Rails.application.config.action_dispatch.cookies_rotations.tap do |cookies|
+    salt = Rails.application.config.action_dispatch.authenticated_encrypted_cookie_salt
+    secret_key_base = Rails.application.secret_key_base
+
+    key_generator = ActiveSupport::KeyGenerator.new(
+      secret_key_base, iterations: 1000, hash_digest_class: OpenSSL::Digest::SHA1
```

Comment:
Why is this SHA1 but the other setting is SHA256?

Reply:
If I understand it well, this is to keep it compatible. On this line it is setting up back the old way (SHA1, https://github.com/rails/rails/blob/1fd0bc1094144b268a3b931a92eb6aeb32fe5c7b/guides/source/configuring.md#configactive_supportkey_generator_hash_digest_class) and later `key_generator_hash_digest_class` is configured to use the new way (SHA256).

That way old cookies are "rotated" when reaches the app without invalidation. They are just migrated from SHA1 to SHA256. We can remove this rotator after some time.